beamer/agent/chain.py

```diff
@@ -632,16 +632,17 @@ def maybe_prove(claim: Claim, context: Context) -> None:
 
     def on_future_done(f: Future) -> None:
         try:
-            f.result()
+            result = f.result()
         except Exception as ex:
             context.logger.error("Optimism prove failed", ex=ex, tx_hash=prove_tx)
         else:
+            timestamp = result.stdout.split()[-1].split(": ")[-1]
```

Comment:
Does this mean, take the last line of the `stdout` and split it by ":" and take the last value from the split?

If so, i would advise to be more explicit and try to search through all the `stdout` lines to find an exact match of the line "Proof timestamp:*". before trying to split it by ":". 
If the `prove-op-message` command changes in a way that it prints anything else after the `Proof timestamp` line then this will break without us noticing.

Reply:
> Does this mean, take the last line of the `stdout` and split it by ":" and take the last value from the split?
> 
> If so, i would advise to be more explicit and try to search through all the `stdout` lines to find an exact match of the line "Proof timestamp:*". before trying to split it by ":". If the `prove-op-message` command changes in a way that it prints anything else after the `Proof timestamp` line then this will break without us noticing.

:100:

Reply:
I would also prefer if this timestamp parsing was handled by the `relayer` Python module, inside `run_relayer_for_tx`. Because that is where the interface to the relayer binary is.